src/test/java/seedu/address/logic/parser/wallet/WalletBudgetCommandParserTest.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.logic.parser.wallet;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_AMOUNT_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_DATE_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.NAME_DESC_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_AMOUNT;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_AMOUNT_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DATE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DATE_AMY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DATE_DESC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.model.transaction.Date.DATE_PATTERN;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.wallet.WalletBudgetCommand;
+import seedu.address.model.transaction.Amount;
+import seedu.address.model.transaction.Budget;
+import seedu.address.model.transaction.Date;
+
+public class WalletBudgetCommandParserTest {
+    private static final String MESSAGE_INVALID_FORMAT =
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, WalletBudgetCommand.MESSAGE_USAGE);
+    private WalletBudgetCommandParser parser = new WalletBudgetCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no arguments
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+
+        // no amount
+        assertParseFailure(parser, PREFIX_DATE + VALID_DATE_AMY, MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidPreamble_failure() {
```

Comment:
I think this is for commands that has <index> only

Reply:
I took preamble to be any string of text, so it fits here